### PR TITLE
feat: Add ability to un-impersonate user after it gets impersonated

### PR DESCRIPTION
**Pull request by martin-trajanovski**
## Description

Add `impersonatingUserId` inside the jwt token to be able to get the information later when you want to un-impersonate and compare if it is correct one because now the endpoint is open for any authorized user.

## Motivation and Context

Previously we were able to only impersonate user as user officer but not go back with one click.

## How Has This Been Tested

- e2e tests
- manual tests

## Fixes

https://jira.esss.lu.se/browse/SWAP-2392

## Changes

https://user-images.githubusercontent.com/6333063/168036262-bbcfcf6b-fd18-4461-b8fc-6abb7d5dfc84.mp4

## Depends on

Frontend PR: https://github.com/UserOfficeProject/user-office-frontend/pull/895

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

File: src/models/User.ts [hunk 1]
```diff
@@ -52,6 +53,7 @@ export class User {
 
 export interface UserWithRole extends User {
   currentRole: Role | undefined;
+  impersonatingUserId?: number | null;
```
Comment (jekabs-karklins) [inline src/models/User.ts:56]: maybe it should be just 
`impersonatingUserId: number | undefined;`
same as `currentRole`

File: src/mutations/UserMutations.ts [hunk 2]
```diff
@@ -408,11 +408,39 @@ export default class UserMutations {
...
+
+    // NOTE: This is checking if person trying to impersonate user is not user officer then reject.
+    if (isImpersonatingUser && !isUserOfficer) {
+      return rejection(
+        'Can not impersonate user because of insufficient permissions',
+        { userId }
+      );
+    }
+
+    if (!isImpersonatingUser && isUserOfficer) {
```
Comment (jekabs-karklins) [inline src/mutations/UserMutations.ts:429]: The message says 'Can not impersonate user with already impersonated user'

Should not the if therefore be `if (isImpersonatingUser && isUserOfficer)` ? or am I missing something

**Comment by simonfernandes** [comment]
I'm sorry @jekabs-karklins, I was reviewing at the same time and left a very similar comment to yours, but accidentally deleted yours instead of mine 🙁

File: src/mutations/UserMutations.ts (hunk 2 above)
Comment (simonfernandes) [inline src/mutations/UserMutations.ts:429]: Comment left by Jekabs which I accidentally deleted:

>The message says 'Can not impersonate user with already impersonated user'
>
>Should not the if therefore be if (isImpersonatingUser && isUserOfficer) ? or am I missing something

Comment (simonfernandes) [reply src/mutations/UserMutations.ts:429]: I was also thinking the same thing, that the boolean would make more sense inverted. 

Could this check also be shortened to just `if (isImpersonatingUser)`, and the same with the user officer check: `if (!isUserOfficer)`?

**Review by simonfernandes — APPROVED** [review]

**Pull request merged**